<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single cell RNA-seq data">
  <meta name="generator" content="bookdown 0.3.3 and GitBook 2.6.7">

  <meta property="og:title" content="Analysis of single cell RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Analysis of single cell RNA-seq data" />
  
  
  

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews, Davis McCarthy and Martin Hemberg">


<meta name="date" content="2016-12-14">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="dealing-with-confounders.html">
<link rel="next" href="clustering-introduction.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- for Facebook -->  
<meta property="og:url" content="http://hemberg-lab.github.io/scRNA.seq.course/" />
<meta property="og:description" content="In this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq. The course is taught through the University of Cambridge Bioinformatics training unit, but the material found on these pages is meant to be used for anyone interested in learning about computational analysis of scRNA-seq data." />
<meta property="og:image" content="http://hemberg-lab.github.io/scRNA.seq.course/figures/RNA-Seq_workflow-5.pdf.jpg" />

<!-- for Twitter -->          
<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:title" content="Analysis of single-cell RNA-seq data" />
<meta name="twitter:description" content="In this course we will be surveying the existing problems as well as the available computational and statistical frameworks available for the analysis of scRNA-seq. The course is taught through the University of Cambridge Bioinformatics training unit, but the material found on these pages is meant to be used for anyone interested in learning about computational analysis of scRNA-seq data." />
<meta name="twitter:image" content="http://hemberg-lab.github.io/scRNA.seq.course/figures/RNA-Seq_workflow-5.pdf.jpg" />

<!-- Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-71525309-1', 'auto');
  ga('send', 'pageview');

</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#video"><i class="fa fa-check"></i><b>1.1</b> Video</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.2</b> Registration</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.3</b> GitHub</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.4</b> License</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.5</b> Prerequisites</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.6</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-requirements.html"><a href="technical-requirements.html#general"><i class="fa fa-check"></i><b>2.1</b> General</a></li>
<li class="chapter" data-level="2.2" data-path="technical-requirements.html"><a href="technical-requirements.html#plotting"><i class="fa fa-check"></i><b>2.2</b> Plotting</a></li>
<li class="chapter" data-level="2.3" data-path="technical-requirements.html"><a href="technical-requirements.html#qc-and-normalisation"><i class="fa fa-check"></i><b>2.3</b> QC and normalisation</a></li>
<li class="chapter" data-level="2.4" data-path="technical-requirements.html"><a href="technical-requirements.html#clustering"><i class="fa fa-check"></i><b>2.4</b> Clustering</a></li>
<li class="chapter" data-level="2.5" data-path="technical-requirements.html"><a href="technical-requirements.html#dropouts"><i class="fa fa-check"></i><b>2.5</b> Dropouts</a></li>
<li class="chapter" data-level="2.6" data-path="technical-requirements.html"><a href="technical-requirements.html#pseudotime"><i class="fa fa-check"></i><b>2.6</b> Pseudotime</a></li>
<li class="chapter" data-level="2.7" data-path="technical-requirements.html"><a href="technical-requirements.html#differential-expression"><i class="fa fa-check"></i><b>2.7</b> Differential Expression</a></li>
<li class="chapter" data-level="2.8" data-path="technical-requirements.html"><a href="technical-requirements.html#extra-tools"><i class="fa fa-check"></i><b>2.8</b> Extra tools</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#protocol"><i class="fa fa-check"></i><b>3.3</b> Protocol</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#computational-analysis"><i class="fa fa-check"></i><b>3.4</b> Computational Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
<li class="chapter" data-level="3.6" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#controls"><i class="fa fa-check"></i><b>3.6</b> Controls</a><ul>
<li class="chapter" data-level="3.6.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#spike-ins"><i class="fa fa-check"></i><b>3.6.1</b> Spike-ins</a></li>
<li class="chapter" data-level="3.6.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#umis"><i class="fa fa-check"></i><b>3.6.2</b> UMIs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>4</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="4.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>4.1</b> Reads QC</a></li>
<li class="chapter" data-level="4.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>4.2</b> Reads alignment</a></li>
<li class="chapter" data-level="4.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>4.3</b> Alignment example</a></li>
<li class="chapter" data-level="4.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>4.4</b> Mapping QC</a></li>
<li class="chapter" data-level="4.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>4.5</b> Reads quantification</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html"><i class="fa fa-check"></i><b>5</b> Unique Molecular Identifiers (UMIs)</a><ul>
<li class="chapter" data-level="5.1" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#mapping-barcodes"><i class="fa fa-check"></i><b>5.2</b> Mapping Barcodes</a></li>
<li class="chapter" data-level="5.3" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#counting-barcodes"><i class="fa fa-check"></i><b>5.3</b> Counting Barcodes</a></li>
<li class="chapter" data-level="5.4" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#correcting-for-errors"><i class="fa fa-check"></i><b>5.4</b> Correcting for Errors</a></li>
<li class="chapter" data-level="5.5" data-path="unique-molecular-identifiers-umis.html"><a href="unique-molecular-identifiers-umis.html#downstream-analysis"><i class="fa fa-check"></i><b>5.5</b> Downstream Analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>6</b> scater package</a><ul>
<li class="chapter" data-level="6.1" data-path="scater-package.html"><a href="scater-package.html#introduction-1"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>6.2</b> scater workflow</a></li>
<li class="chapter" data-level="6.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>6.3</b> Terminology</a></li>
<li class="chapter" data-level="6.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>6.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>7</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="7.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-2"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>7.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="7.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>7.3</b> Cell QC</a><ul>
<li class="chapter" data-level="7.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>7.3.1</b> Library size</a></li>
<li class="chapter" data-level="7.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>7.3.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="7.3.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>7.3.3</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>7.4</b> Cell filtering</a><ul>
<li class="chapter" data-level="7.4.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>7.4.1</b> Manual</a></li>
<li class="chapter" data-level="7.4.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default-thresholds"><i class="fa fa-check"></i><b>7.4.2</b> Default Thresholds</a></li>
<li class="chapter" data-level="7.4.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>7.4.3</b> Automatic</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>7.5</b> Compare filterings</a></li>
<li class="chapter" data-level="7.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-analysis"><i class="fa fa-check"></i><b>7.6</b> Gene analysis</a><ul>
<li class="chapter" data-level="7.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>7.6.1</b> Gene expression</a></li>
<li class="chapter" data-level="7.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>7.6.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>7.7</b> Save the data</a></li>
<li class="chapter" data-level="7.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#big-exercise"><i class="fa fa-check"></i><b>7.8</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>8</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="9" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>9</b> Data visualization</a><ul>
<li class="chapter" data-level="9.1" data-path="data-visualization.html"><a href="data-visualization.html#introduction-3"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="data-visualization.html"><a href="data-visualization.html#visual-pca"><i class="fa fa-check"></i><b>9.2</b> PCA plot</a><ul>
<li class="chapter" data-level="9.2.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc"><i class="fa fa-check"></i><b>9.2.1</b> Before QC</a></li>
<li class="chapter" data-level="9.2.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc"><i class="fa fa-check"></i><b>9.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="data-visualization.html"><a href="data-visualization.html#visual-tsne"><i class="fa fa-check"></i><b>9.3</b> tSNE map</a><ul>
<li class="chapter" data-level="9.3.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc-1"><i class="fa fa-check"></i><b>9.3.1</b> Before QC</a></li>
<li class="chapter" data-level="9.3.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc-1"><i class="fa fa-check"></i><b>9.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="data-visualization.html"><a href="data-visualization.html#big-exercise-1"><i class="fa fa-check"></i><b>9.4</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="data-visualization-reads.html"><a href="data-visualization-reads.html"><i class="fa fa-check"></i><b>10</b> Data visualization (Reads)</a></li>
<li class="chapter" data-level="11" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>11</b> Confounding factors</a><ul>
<li class="chapter" data-level="11.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-4"><i class="fa fa-check"></i><b>11.1</b> Introduction</a></li>
<li class="chapter" data-level="11.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>11.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="11.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>11.2.1</b> Detected genes</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>11.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="11.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>11.4</b> Other confounders</a></li>
<li class="chapter" data-level="11.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise"><i class="fa fa-check"></i><b>11.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>12</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="13" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>13</b> Normalization for library size</a><ul>
<li class="chapter" data-level="13.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-5"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>13.2</b> Library size</a></li>
<li class="chapter" data-level="13.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>13.3</b> Normalisations</a><ul>
<li class="chapter" data-level="13.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>13.3.1</b> Raw</a></li>
<li class="chapter" data-level="13.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>13.3.2</b> CPM</a></li>
<li class="chapter" data-level="13.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>13.3.3</b> TMM</a></li>
<li class="chapter" data-level="13.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#scran"><i class="fa fa-check"></i><b>13.3.4</b> scran</a></li>
<li class="chapter" data-level="13.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#size-factor-rle"><i class="fa fa-check"></i><b>13.3.5</b> Size-factor (RLE)</a></li>
<li class="chapter" data-level="13.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>13.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#downsampling"><i class="fa fa-check"></i><b>13.4</b> Downsampling</a></li>
<li class="chapter" data-level="13.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalizing-for-genetranscript-length"><i class="fa fa-check"></i><b>13.5</b> Normalizing for gene/transcript length</a></li>
<li class="chapter" data-level="13.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-1"><i class="fa fa-check"></i><b>13.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>14</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="15" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html"><i class="fa fa-check"></i><b>15</b> Dealing with confounders</a><ul>
<li class="chapter" data-level="15.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#introduction-6"><i class="fa fa-check"></i><b>15.1</b> Introduction</a></li>
<li class="chapter" data-level="15.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>15.2</b> Remove Unwanted Variation</a><ul>
<li class="chapter" data-level="15.2.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#ruvg"><i class="fa fa-check"></i><b>15.2.1</b> RUVg</a></li>
<li class="chapter" data-level="15.2.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#ruvs"><i class="fa fa-check"></i><b>15.2.2</b> RUVs</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-1"><i class="fa fa-check"></i><b>15.3</b> Effectiveness 1</a></li>
<li class="chapter" data-level="15.4" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-2"><i class="fa fa-check"></i><b>15.4</b> Effectiveness 2</a></li>
<li class="chapter" data-level="15.5" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#effectiveness-3"><i class="fa fa-check"></i><b>15.5</b> Effectiveness 3</a><ul>
<li class="chapter" data-level="15.5.1" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-raw-counts"><i class="fa fa-check"></i><b>15.5.1</b> DE (raw counts)</a></li>
<li class="chapter" data-level="15.5.2" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-ruvg-k-2"><i class="fa fa-check"></i><b>15.5.2</b> DE (RUVg, k = 2)</a></li>
<li class="chapter" data-level="15.5.3" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#de-ruvs-k-2"><i class="fa fa-check"></i><b>15.5.3</b> DE (RUVs, k = 2)</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="dealing-with-confounders.html"><a href="dealing-with-confounders.html#exercise-2"><i class="fa fa-check"></i><b>15.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html"><i class="fa fa-check"></i><b>16</b> Dealing with confounders (Reads)</a><ul>
<li class="chapter" data-level="16.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#remove-unwanted-variation-1"><i class="fa fa-check"></i><b>16.1</b> Remove Unwanted Variation</a><ul>
<li class="chapter" data-level="16.1.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#ruvg-1"><i class="fa fa-check"></i><b>16.1.1</b> RUVg</a></li>
<li class="chapter" data-level="16.1.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#ruvs-1"><i class="fa fa-check"></i><b>16.1.2</b> RUVs</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-1-1"><i class="fa fa-check"></i><b>16.2</b> Effectiveness 1</a></li>
<li class="chapter" data-level="16.3" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-2-1"><i class="fa fa-check"></i><b>16.3</b> Effectiveness 2</a></li>
<li class="chapter" data-level="16.4" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#effectiveness-3-1"><i class="fa fa-check"></i><b>16.4</b> Effectiveness 3</a><ul>
<li class="chapter" data-level="16.4.1" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-raw-counts-1"><i class="fa fa-check"></i><b>16.4.1</b> DE (raw counts)</a></li>
<li class="chapter" data-level="16.4.2" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-ruvg-k-2-1"><i class="fa fa-check"></i><b>16.4.2</b> DE (RUVg, k = 2)</a></li>
<li class="chapter" data-level="16.4.3" data-path="dealing-with-confounders-reads.html"><a href="dealing-with-confounders-reads.html#de-ruvs-k-2-1"><i class="fa fa-check"></i><b>16.4.3</b> DE (RUVs, k = 2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="clustering-introduction.html"><a href="clustering-introduction.html"><i class="fa fa-check"></i><b>17</b> Clustering Introduction</a><ul>
<li class="chapter" data-level="17.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#introduction-7"><i class="fa fa-check"></i><b>17.1</b> Introduction</a></li>
<li class="chapter" data-level="17.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#dimensionality-reductions"><i class="fa fa-check"></i><b>17.2</b> Dimensionality reductions</a><ul>
<li class="chapter" data-level="17.2.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#pca"><i class="fa fa-check"></i><b>17.2.1</b> PCA</a></li>
<li class="chapter" data-level="17.2.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#spectral"><i class="fa fa-check"></i><b>17.2.2</b> Spectral</a></li>
<li class="chapter" data-level="17.2.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tsne"><i class="fa fa-check"></i><b>17.2.3</b> tSNE</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#clustering-methods"><i class="fa fa-check"></i><b>17.3</b> Clustering methods</a><ul>
<li class="chapter" data-level="17.3.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#hierarchical-clustering"><i class="fa fa-check"></i><b>17.3.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="17.3.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#k-means"><i class="fa fa-check"></i><b>17.3.2</b> k-means</a></li>
<li class="chapter" data-level="17.3.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#graph-based-methods"><i class="fa fa-check"></i><b>17.3.3</b> Graph-based methods</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="clustering-introduction.html"><a href="clustering-introduction.html#challenges-in-clustering"><i class="fa fa-check"></i><b>17.4</b> Challenges in clustering</a></li>
<li class="chapter" data-level="17.5" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tools-for-scrna-seq-data"><i class="fa fa-check"></i><b>17.5</b> Tools for scRNA-seq data</a><ul>
<li class="chapter" data-level="17.5.1" data-path="clustering-introduction.html"><a href="clustering-introduction.html#sincera"><i class="fa fa-check"></i><b>17.5.1</b> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></a></li>
<li class="chapter" data-level="17.5.2" data-path="clustering-introduction.html"><a href="clustering-introduction.html#pcareduce"><i class="fa fa-check"></i><b>17.5.2</b> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></a></li>
<li class="chapter" data-level="17.5.3" data-path="clustering-introduction.html"><a href="clustering-introduction.html#sc3"><i class="fa fa-check"></i><b>17.5.3</b> <a href="http://bioconductor.org/packages/SC3/">SC3</a></a></li>
<li class="chapter" data-level="17.5.4" data-path="clustering-introduction.html"><a href="clustering-introduction.html#tsne-k-means"><i class="fa fa-check"></i><b>17.5.4</b> tSNE + k-means</a></li>
<li class="chapter" data-level="17.5.5" data-path="clustering-introduction.html"><a href="clustering-introduction.html#seurat"><i class="fa fa-check"></i><b>17.5.5</b> <a href="https://github.com/Puriney/seurat">SEURAT</a></a></li>
<li class="chapter" data-level="17.5.6" data-path="clustering-introduction.html"><a href="clustering-introduction.html#snn-cliq"><i class="fa fa-check"></i><b>17.5.6</b> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="clust-methods.html"><a href="clust-methods.html"><i class="fa fa-check"></i><b>18</b> Clustering example</a><ul>
<li class="chapter" data-level="18.1" data-path="clust-methods.html"><a href="clust-methods.html#patient-dataset"><i class="fa fa-check"></i><b>18.1</b> Patient dataset</a></li>
<li class="chapter" data-level="18.2" data-path="clust-methods.html"><a href="clust-methods.html#sc3-1"><i class="fa fa-check"></i><b>18.2</b> SC3</a></li>
<li class="chapter" data-level="18.3" data-path="clust-methods.html"><a href="clust-methods.html#pcareduce-1"><i class="fa fa-check"></i><b>18.3</b> pcaReduce</a></li>
<li class="chapter" data-level="18.4" data-path="clust-methods.html"><a href="clust-methods.html#tsne-kmeans"><i class="fa fa-check"></i><b>18.4</b> tSNE + kmeans</a></li>
<li class="chapter" data-level="18.5" data-path="clust-methods.html"><a href="clust-methods.html#snn-cliq-1"><i class="fa fa-check"></i><b>18.5</b> SNN-Cliq</a></li>
<li class="chapter" data-level="18.6" data-path="clust-methods.html"><a href="clust-methods.html#sincera-1"><i class="fa fa-check"></i><b>18.6</b> SINCERA</a></li>
<li class="chapter" data-level="18.7" data-path="clust-methods.html"><a href="clust-methods.html#seurat-1"><i class="fa fa-check"></i><b>18.7</b> SEURAT</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html"><i class="fa fa-check"></i><b>19</b> Identification of important genes</a><ul>
<li class="chapter" data-level="19.1" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#fitting-the-models"><i class="fa fa-check"></i><b>19.1</b> Fitting the models</a></li>
<li class="chapter" data-level="19.2" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#the-michaelis-menten-equation"><i class="fa fa-check"></i><b>19.2</b> The Michaelis-Menten Equation</a></li>
<li class="chapter" data-level="19.3" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#right-outliers"><i class="fa fa-check"></i><b>19.3</b> Right outliers</a></li>
<li class="chapter" data-level="19.4" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#validation-of-de-results"><i class="fa fa-check"></i><b>19.4</b> Validation of DE results</a></li>
<li class="chapter" data-level="19.5" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#comparing-m3drop-to-other-methods"><i class="fa fa-check"></i><b>19.5</b> Comparing M3Drop to other methods</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html"><i class="fa fa-check"></i><b>20</b> Pseudotime analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#tscan"><i class="fa fa-check"></i><b>20.1</b> TSCAN</a></li>
<li class="chapter" data-level="20.2" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#monocle"><i class="fa fa-check"></i><b>20.2</b> monocle</a></li>
<li class="chapter" data-level="20.3" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#diffusion-maps"><i class="fa fa-check"></i><b>20.3</b> Diffusion maps</a></li>
<li class="chapter" data-level="20.4" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>20.4</b> Comparison of the methods</a></li>
<li class="chapter" data-level="20.5" data-path="pseudotime-analysis.html"><a href="pseudotime-analysis.html#expression-of-genes-through-time"><i class="fa fa-check"></i><b>20.5</b> Expression of genes through time</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html"><i class="fa fa-check"></i><b>21</b> Differential Expression (DE) analysis</a><ul>
<li class="chapter" data-level="21.1" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#bulk-rna-seq-1"><i class="fa fa-check"></i><b>21.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="21.2" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#single-cell-rna-seq"><i class="fa fa-check"></i><b>21.2</b> Single cell RNA-seq</a></li>
<li class="chapter" data-level="21.3" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#scrna-seq-synthetic-data"><i class="fa fa-check"></i><b>21.3</b> scRNA-seq synthetic data</a></li>
<li class="chapter" data-level="21.4" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#single-gene-expression"><i class="fa fa-check"></i><b>21.4</b> Single gene expression</a></li>
<li class="chapter" data-level="21.5" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#poisson-beta-distribution"><i class="fa fa-check"></i><b>21.5</b> Poisson-Beta distribution</a></li>
<li class="chapter" data-level="21.6" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#sample-size"><i class="fa fa-check"></i><b>21.6</b> Sample size</a></li>
<li class="chapter" data-level="21.7" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#dropout-noise"><i class="fa fa-check"></i><b>21.7</b> Dropout noise</a></li>
<li class="chapter" data-level="21.8" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#dispersion-noise"><i class="fa fa-check"></i><b>21.8</b> Dispersion noise</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html"><i class="fa fa-check"></i><b>22</b> DE in a synthetic dataset</a><ul>
<li class="chapter" data-level="22.1" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#dataset"><i class="fa fa-check"></i><b>22.1</b> Dataset</a></li>
<li class="chapter" data-level="22.2" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#de-in-scrna-seq"><i class="fa fa-check"></i><b>22.2</b> DE in scRNA-seq</a></li>
<li class="chapter" data-level="22.3" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#kolmogorov-smirnov-test"><i class="fa fa-check"></i><b>22.3</b> Kolmogorov-Smirnov test</a></li>
<li class="chapter" data-level="22.4" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#performance-of-ks-test"><i class="fa fa-check"></i><b>22.4</b> Performance of KS test</a></li>
<li class="chapter" data-level="22.5" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#deseq2"><i class="fa fa-check"></i><b>22.5</b> DESeq2</a></li>
<li class="chapter" data-level="22.6" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#scde"><i class="fa fa-check"></i><b>22.6</b> SCDE</a></li>
<li class="chapter" data-level="22.7" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#comparison-of-the-methods-1"><i class="fa fa-check"></i><b>22.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="22.8" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#beyond-changes-in-the-mean"><i class="fa fa-check"></i><b>22.8</b> Beyond changes in the mean</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html"><i class="fa fa-check"></i><b>23</b> DE in a real dataset</a><ul>
<li class="chapter" data-level="23.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#introduction-8"><i class="fa fa-check"></i><b>23.1</b> Introduction</a></li>
<li class="chapter" data-level="23.2" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#ks-test"><i class="fa fa-check"></i><b>23.2</b> KS-test</a></li>
<li class="chapter" data-level="23.3" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#deseq2-1"><i class="fa fa-check"></i><b>23.3</b> DESeq2</a></li>
<li class="chapter" data-level="23.4" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#scde-1"><i class="fa fa-check"></i><b>23.4</b> SCDE</a></li>
<li class="chapter" data-level="23.5" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#comparison-of-the-methods-2"><i class="fa fa-check"></i><b>23.5</b> Comparison of the methods</a></li>
<li class="chapter" data-level="23.6" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#visualisation"><i class="fa fa-check"></i><b>23.6</b> Visualisation</a><ul>
<li class="chapter" data-level="23.6.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#volcano-plots"><i class="fa fa-check"></i><b>23.6.1</b> Volcano plots</a></li>
<li class="chapter" data-level="23.6.2" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#ma-plot"><i class="fa fa-check"></i><b>23.6.2</b> MA-plot</a></li>
<li class="chapter" data-level="23.6.3" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#heatmap-of-de-genes"><i class="fa fa-check"></i><b>23.6.3</b> Heatmap of DE genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><i class="fa fa-check"></i><b>24</b> “Ideal” scRNAseq pipeline (as of Oct 2016)</a><ul>
<li class="chapter" data-level="24.1" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#experimental-design"><i class="fa fa-check"></i><b>24.1</b> Experimental Design</a></li>
<li class="chapter" data-level="24.2" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#processing-reads"><i class="fa fa-check"></i><b>24.2</b> Processing Reads</a></li>
<li class="chapter" data-level="24.3" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#preparing-expression-matrix"><i class="fa fa-check"></i><b>24.3</b> Preparing Expression Matrix</a></li>
<li class="chapter" data-level="24.4" data-path="ideal-scrnaseq-pipeline-as-of-oct-2016.html"><a href="ideal-scrnaseq-pipeline-as-of-oct-2016.html#biological-interpretation"><i class="fa fa-check"></i><b>24.4</b> Biological Interpretation</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="advanced-exercises.html"><a href="advanced-exercises.html"><i class="fa fa-check"></i><b>25</b> Advanced exercises</a></li>
<li class="chapter" data-level="26" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>26</b> Resources</a><ul>
<li class="chapter" data-level="26.1" data-path="resources.html"><a href="resources.html#scrna-seq-protocols"><i class="fa fa-check"></i><b>26.1</b> scRNA-seq protocols</a></li>
<li class="chapter" data-level="26.2" data-path="resources.html"><a href="resources.html#external-rna-control-consortium-ercc"><i class="fa fa-check"></i><b>26.2</b> External RNA Control Consortium (ERCC)</a></li>
<li class="chapter" data-level="26.3" data-path="resources.html"><a href="resources.html#scrna-seq-analysis-tools"><i class="fa fa-check"></i><b>26.3</b> scRNA-seq analysis tools</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>27</b> References</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dealing-with-confounders-reads" class="section level1">
<h1><span class="header-section-number">16</span> Dealing with confounders (Reads)</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scRNA.seq.funcs)
<span class="kw">library</span>(RUVSeq)
<span class="kw">library</span>(scater, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)
<span class="kw">library</span>(scran)
<span class="kw">library</span>(edgeR)
<span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
reads &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;blischak/reads.rds&quot;</span>)
reads.qc &lt;-<span class="st"> </span>reads[<span class="kw">fData</span>(reads)$use, <span class="kw">pData</span>(reads)$use]
endog_genes &lt;-<span class="st"> </span>!<span class="kw">fData</span>(reads.qc)$is_feature_control
erccs &lt;-<span class="st"> </span><span class="kw">fData</span>(reads.qc)$is_feature_control</code></pre></div>
<div id="remove-unwanted-variation-1" class="section level2">
<h2><span class="header-section-number">16.1</span> Remove Unwanted Variation</h2>
<div id="ruvg-1" class="section level3">
<h3><span class="header-section-number">16.1.1</span> RUVg</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ruvg &lt;-<span class="st"> </span><span class="kw">RUVg</span>(<span class="kw">counts</span>(reads.qc), erccs, <span class="dt">k =</span> <span class="dv">1</span>)
<span class="kw">set_exprs</span>(reads.qc, <span class="st">&quot;ruvg1&quot;</span>) &lt;-<span class="st"> </span>ruvg$normalizedCounts
ruvg &lt;-<span class="st"> </span><span class="kw">RUVg</span>(<span class="kw">counts</span>(reads.qc), erccs, <span class="dt">k =</span> <span class="dv">2</span>)
<span class="kw">set_exprs</span>(reads.qc, <span class="st">&quot;ruvg2&quot;</span>) &lt;-<span class="st"> </span>ruvg$normalizedCounts
<span class="kw">set_exprs</span>(reads.qc, <span class="st">&quot;ruvg2_logcpm&quot;</span>) &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">t</span>(<span class="kw">t</span>(ruvg$normalizedCounts) /<span class="st"> </span>
<span class="st">                                           </span><span class="kw">colSums</span>(ruvg$normalizedCounts)) +<span class="st"> </span><span class="dv">1</span>)</code></pre></div>
</div>
<div id="ruvs-1" class="section level3">
<h3><span class="header-section-number">16.1.2</span> RUVs</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scIdx &lt;-<span class="st"> </span><span class="kw">matrix</span>(-<span class="dv">1</span>, <span class="dt">ncol =</span> <span class="kw">max</span>(<span class="kw">table</span>(reads.qc$individual)), <span class="dt">nrow =</span> <span class="dv">3</span>)
tmp &lt;-<span class="st"> </span><span class="kw">which</span>(reads.qc$individual ==<span class="st"> &quot;NA19098&quot;</span>)
scIdx[<span class="dv">1</span>, <span class="dv">1</span>:<span class="kw">length</span>(tmp)] &lt;-<span class="st"> </span>tmp
tmp &lt;-<span class="st"> </span><span class="kw">which</span>(reads.qc$individual ==<span class="st"> &quot;NA19101&quot;</span>)
scIdx[<span class="dv">2</span>, <span class="dv">1</span>:<span class="kw">length</span>(tmp)] &lt;-<span class="st"> </span>tmp
tmp &lt;-<span class="st"> </span><span class="kw">which</span>(reads.qc$individual ==<span class="st"> &quot;NA19239&quot;</span>)
scIdx[<span class="dv">3</span>, <span class="dv">1</span>:<span class="kw">length</span>(tmp)] &lt;-<span class="st"> </span>tmp
cIdx &lt;-<span class="st"> </span><span class="kw">rownames</span>(reads.qc)
ruvs &lt;-<span class="st"> </span><span class="kw">RUVs</span>(<span class="kw">counts</span>(reads.qc), cIdx, <span class="dt">k =</span> <span class="dv">1</span>, <span class="dt">scIdx =</span> scIdx, <span class="dt">isLog =</span> <span class="ot">FALSE</span>)
<span class="kw">set_exprs</span>(reads.qc, <span class="st">&quot;ruvs1&quot;</span>) &lt;-<span class="st"> </span>ruvs$normalizedCounts
ruvs &lt;-<span class="st"> </span><span class="kw">RUVs</span>(<span class="kw">counts</span>(reads.qc), cIdx, <span class="dt">k =</span> <span class="dv">2</span>, <span class="dt">scIdx =</span> scIdx, <span class="dt">isLog =</span> <span class="ot">FALSE</span>)
<span class="kw">set_exprs</span>(reads.qc, <span class="st">&quot;ruvs2&quot;</span>) &lt;-<span class="st"> </span>ruvs$normalizedCounts
<span class="kw">set_exprs</span>(reads.qc, <span class="st">&quot;ruvs2_logcpm&quot;</span>) &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">t</span>(<span class="kw">t</span>(ruvs$normalizedCounts) /<span class="st"> </span>
<span class="st">                                           </span><span class="kw">colSums</span>(ruvs$normalizedCounts)) +<span class="st"> </span><span class="dv">1</span>)</code></pre></div>
</div>
</div>
<div id="effectiveness-1-1" class="section level2">
<h2><span class="header-section-number">16.2</span> Effectiveness 1</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotPCA</span>(
    reads.qc[endog_genes, ],
    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
    <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
    <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
    <span class="dt">exprs_values =</span> <span class="st">&quot;ruvg1&quot;</span>) +
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA - RUVg normalisation: k = 1&quot;</span>)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotPCA</span>(
    reads.qc[endog_genes, ],
    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
    <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
    <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
    <span class="dt">exprs_values =</span> <span class="st">&quot;ruvg2&quot;</span>) +
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA - RUVg normalisation: k = 2&quot;</span>)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotPCA</span>(
    reads.qc[endog_genes, ],
    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
    <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
    <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
    <span class="dt">exprs_values =</span> <span class="st">&quot;ruvs1&quot;</span>) +
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA - RUVs normalisation: k = 1&quot;</span>)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotPCA</span>(
    reads.qc[endog_genes, ],
    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
    <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
    <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
    <span class="dt">exprs_values =</span> <span class="st">&quot;ruvs2&quot;</span>) +
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA - RUVs normalisation: k = 2&quot;</span>)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotPCA</span>(
    reads.qc[endog_genes, ],
    <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
    <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
    <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
    <span class="dt">exprs_values =</span> <span class="st">&quot;ruvs2_logcpm&quot;</span>) +
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;PCA - RUVs normalisation log2-cpm: k = 2&quot;</span>)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="effectiveness-2-1" class="section level2">
<h2><span class="header-section-number">16.3</span> Effectiveness 2</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(
    <span class="kw">list</span>(
        <span class="st">&quot;Raw counts&quot;</span> =<span class="st"> </span><span class="kw">calc_cell_RLE</span>(<span class="kw">counts</span>(reads.qc), erccs),
        <span class="st">&quot;RUVg (k = 1)&quot;</span> =<span class="st"> </span><span class="kw">calc_cell_RLE</span>(<span class="kw">assayData</span>(reads.qc)$ruvg1, erccs),
        <span class="st">&quot;RUVg (k = 2)&quot;</span> =<span class="st"> </span><span class="kw">calc_cell_RLE</span>(<span class="kw">assayData</span>(reads.qc)$ruvg2, erccs),
        <span class="st">&quot;RUVs (k = 1)&quot;</span> =<span class="st"> </span><span class="kw">calc_cell_RLE</span>(<span class="kw">assayData</span>(reads.qc)$ruvs1, erccs),
        <span class="st">&quot;RUVs (k = 2)&quot;</span> =<span class="st"> </span><span class="kw">calc_cell_RLE</span>(<span class="kw">assayData</span>(reads.qc)$ruvs2, erccs)
    )
)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="effectiveness-3-1" class="section level2">
<h2><span class="header-section-number">16.4</span> Effectiveness 3</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keep &lt;-<span class="st"> </span><span class="kw">c</span>(
    <span class="kw">sample</span>(<span class="kw">which</span>(reads.qc$batch ==<span class="st"> &quot;NA19101.r1&quot;</span>), <span class="dv">20</span>), 
    <span class="kw">sample</span>(<span class="kw">which</span>(reads.qc$batch ==<span class="st"> &quot;NA19101.r2&quot;</span>), <span class="dv">20</span>),
    <span class="kw">sample</span>(<span class="kw">which</span>(reads.qc$batch ==<span class="st"> &quot;NA19101.r3&quot;</span>), <span class="dv">20</span>)
)
design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(~reads.qc[, keep]$batch)</code></pre></div>
<div id="de-raw-counts-1" class="section level3">
<h3><span class="header-section-number">16.4.1</span> DE (raw counts)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dge1 &lt;-<span class="st"> </span><span class="kw">DGEList</span>(
    <span class="dt">counts =</span> <span class="kw">counts</span>(reads.qc[, keep]), 
    <span class="dt">norm.factors =</span> <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">length</span>(keep)),
    <span class="dt">group =</span> reads.qc[, keep]$batch
)
dge1 &lt;-<span class="st"> </span><span class="kw">estimateDisp</span>(dge1, <span class="dt">design =</span> design, <span class="dt">trend.method =</span> <span class="st">&quot;none&quot;</span>)
<span class="kw">plotBCV</span>(dge1)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit1 &lt;-<span class="st"> </span><span class="kw">glmFit</span>(dge1, design)
res1 &lt;-<span class="st"> </span><span class="kw">glmLRT</span>(fit1)
<span class="kw">topTags</span>(res1)</code></pre></div>
<pre><code>## Coefficient:  reads.qc[, keep]$batchNA19101.r3 
##                     logFC    logCPM       LR       PValue         FDR
## ENSG00000165115 -7.792113 2.3887218 26.19783 3.081679e-07 0.002884342
## ENSG00000140280  7.300312 1.6569522 25.85753 3.675684e-07 0.002884342
## ENSG00000105877 -6.966115 1.4022300 24.36423 7.973575e-07 0.002884342
## ENSG00000155542  7.637098 2.0975586 23.91593 1.006354e-06 0.002884342
## ENSG00000087116 -5.697505 0.6201158 23.87020 1.030550e-06 0.002884342
## ENSG00000144476  7.094743 1.5421854 23.78440 1.077520e-06 0.002884342
## ENSG00000177483 -5.788801 0.6504550 23.05738 1.572381e-06 0.002949070
## ENSG00000165105 -6.407054 1.0143121 22.94208 1.669570e-06 0.002949070
## ENSG00000057704 -7.953420 2.2337620 22.72356 1.870600e-06 0.002949070
## ENSG00000105642  6.306497 0.9597862 22.47384 2.130253e-06 0.002949070</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw">decideTestsDGE</span>(res1))</code></pre></div>
<pre><code>##    [,1] 
## -1   541
## 0  14783
## 1    737</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSmear</span>(
    res1, <span class="dt">lowess =</span> <span class="ot">TRUE</span>,
    <span class="dt">de.tags =</span> <span class="kw">rownames</span>(<span class="kw">topTags</span>(res1, <span class="dt">n =</span> <span class="kw">sum</span>(<span class="kw">abs</span>(<span class="kw">decideTestsDGE</span>(res1))))$table)
)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="de-ruvg-k-2-1" class="section level3">
<h3><span class="header-section-number">16.4.2</span> DE (RUVg, k = 2)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design_ruvg &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(~ruvg$W[keep,] +<span class="st"> </span>reads.qc[, keep]$batch)
<span class="kw">head</span>(design_ruvg)</code></pre></div>
<pre><code>##   (Intercept) ruvg$W[keep, ]W_1 ruvg$W[keep, ]W_2
## 1           1       -0.01544610        0.04957296
## 2           1        0.01829566        0.06354737
## 3           1        0.01814940       -0.06327037
## 4           1       -0.02585604        0.02753199
## 5           1       -0.01483137        0.05209915
## 6           1        0.01188452        0.06237138
##   reads.qc[, keep]$batchNA19101.r2 reads.qc[, keep]$batchNA19101.r3
## 1                                0                                0
## 2                                0                                0
## 3                                0                                0
## 4                                0                                0
## 5                                0                                0
## 6                                0                                0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dge_ruvg &lt;-<span class="st"> </span><span class="kw">estimateDisp</span>(dge1, <span class="dt">design =</span> design_ruvg, <span class="dt">trend.method =</span> <span class="st">&quot;none&quot;</span>)
<span class="kw">plotBCV</span>(dge_ruvg)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit2 &lt;-<span class="st"> </span><span class="kw">glmFit</span>(dge_ruvg, design_ruvg)
res2 &lt;-<span class="st"> </span><span class="kw">glmLRT</span>(fit2)
<span class="kw">topTags</span>(res2)</code></pre></div>
<pre><code>## Coefficient:  reads.qc[, keep]$batchNA19101.r3 
##                     logFC    logCPM       LR       PValue          FDR
## ENSG00000143127  6.393263 1.3905478 27.77041 1.365996e-07 0.0008543374
## ENSG00000151014 -7.739377 1.4502069 26.67914 2.402013e-07 0.0008543374
## ENSG00000204538 -5.703323 0.6028395 26.50373 2.630294e-07 0.0008543374
## ENSG00000165959 -8.137890 2.2748771 26.16507 3.134420e-07 0.0008543374
## ENSG00000083844 -7.456655 1.0307443 26.04695 3.332154e-07 0.0008543374
## ENSG00000140280  7.452433 1.6571621 25.84269 3.704051e-07 0.0008543374
## ENSG00000165698 -6.119602 0.8832588 25.19172 5.190466e-07 0.0008543374
## ENSG00000105877 -6.746347 1.4023211 25.05371 5.575546e-07 0.0008543374
## ENSG00000153208  8.123110 2.7499421 24.97353 5.812274e-07 0.0008543374
## ENSG00000161381  6.157689 1.0884471 24.95866 5.857296e-07 0.0008543374</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw">decideTestsDGE</span>(res2))</code></pre></div>
<pre><code>##    [,1] 
## -1   252
## 0  15547
## 1    262</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSmear</span>(
    res2, <span class="dt">lowess =</span> <span class="ot">TRUE</span>,
    <span class="dt">de.tags =</span> <span class="kw">rownames</span>(<span class="kw">topTags</span>(res2, <span class="dt">n =</span> <span class="kw">sum</span>(<span class="kw">abs</span>(<span class="kw">decideTestsDGE</span>(res2))))$table)
)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="de-ruvs-k-2-1" class="section level3">
<h3><span class="header-section-number">16.4.3</span> DE (RUVs, k = 2)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design_ruvs &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(~ruvs$W[keep,] +<span class="st"> </span>reads.qc[, keep]$batch)
<span class="kw">head</span>(design_ruvs)</code></pre></div>
<pre><code>##   (Intercept) ruvs$W[keep, ]W_1 ruvs$W[keep, ]W_2
## 1           1         0.3093128         0.1938905
## 2           1         0.2871249         0.2263144
## 3           1         0.2702873         0.2169784
## 4           1         0.2879353         0.1807737
## 5           1         0.2947330         0.1690802
## 6           1         0.3544075         0.2425527
##   reads.qc[, keep]$batchNA19101.r2 reads.qc[, keep]$batchNA19101.r3
## 1                                0                                0
## 2                                0                                0
## 3                                0                                0
## 4                                0                                0
## 5                                0                                0
## 6                                0                                0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dge_ruvs &lt;-<span class="st"> </span><span class="kw">estimateDisp</span>(dge1, <span class="dt">design =</span> design_ruvs, <span class="dt">trend.method =</span> <span class="st">&quot;none&quot;</span>)
<span class="kw">plotBCV</span>(dge_ruvs)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit3 &lt;-<span class="st"> </span><span class="kw">glmFit</span>(dge_ruvs, design_ruvs)
res3 &lt;-<span class="st"> </span><span class="kw">glmLRT</span>(fit3)
<span class="kw">topTags</span>(res3)</code></pre></div>
<pre><code>## Coefficient:  reads.qc[, keep]$batchNA19101.r3 
##                     logFC    logCPM       LR       PValue          FDR
## ENSG00000111424 -6.201651 0.8462288 46.63224 8.563976e-12 5.114170e-08
## ENSG00000197980 -6.186708 0.8368680 46.52578 9.042109e-12 5.114170e-08
## ENSG00000154227 -6.170589 0.8274457 46.41815 9.552649e-12 5.114170e-08
## ENSG00000138823 -5.642913 0.5232302 42.63487 6.597285e-11 2.648975e-07
## ENSG00000267882 -4.751404 0.1288729 36.21474 1.767284e-09 5.676868e-06
## ENSG00000117152  6.604074 1.1446503 33.76140 6.230290e-09 1.667745e-05
## ENSG00000184564  6.232844 0.8503488 32.35087 1.286993e-08 2.952913e-05
## ENSG00000165046 -5.249966 0.4359222 30.27018 3.758596e-08 7.446001e-05
## ENSG00000167759 -6.817094 1.3172810 29.99466 4.332372e-08 7.446001e-05
## ENSG00000165115 -8.845112 2.3887858 29.86329 4.636076e-08 7.446001e-05</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw">decideTestsDGE</span>(res3))</code></pre></div>
<pre><code>##    [,1] 
## -1   334
## 0  15368
## 1    359</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSmear</span>(
    res3, <span class="dt">lowess =</span> <span class="ot">TRUE</span>,
    <span class="dt">de.tags =</span> <span class="kw">rownames</span>(<span class="kw">topTags</span>(res3, <span class="dt">n =</span> <span class="kw">sum</span>(<span class="kw">abs</span>(<span class="kw">decideTestsDGE</span>(res3))))$table)
)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">reads.qc &lt;-<span class="st"> </span>scran::<span class="kw">computeSumFactors</span>(reads.qc, <span class="dt">sizes =</span> <span class="dv">15</span>)
dge_ruvs$samples$norm.factors &lt;-<span class="st"> </span><span class="kw">sizeFactors</span>(reads.qc)[keep]
dge_ruvs_sf &lt;-<span class="st"> </span><span class="kw">estimateDisp</span>(dge_ruvs, <span class="dt">design =</span> design_ruvs, <span class="dt">trend.method =</span> <span class="st">&quot;none&quot;</span>)
<span class="kw">plotBCV</span>(dge_ruvs_sf)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit4 &lt;-<span class="st"> </span><span class="kw">glmFit</span>(dge_ruvs_sf, design_ruvs)
res4 &lt;-<span class="st"> </span><span class="kw">glmLRT</span>(fit4)
<span class="kw">topTags</span>(res4)</code></pre></div>
<pre><code>## Coefficient:  reads.qc[, keep]$batchNA19101.r3 
##                     logFC       logCPM       LR       PValue          FDR
## ENSG00000111424 -6.281451  0.746154188 45.81432 1.300094e-11 7.757951e-08
## ENSG00000197980 -6.266449  0.736500540 45.70864 1.372164e-11 7.757951e-08
## ENSG00000154227 -6.250268  0.726781398 45.60179 1.449091e-11 7.757951e-08
## ENSG00000138823 -5.720215  0.411697275 41.84588 9.875881e-11 3.965413e-07
## ENSG00000267882 -4.824953 -0.001287811 35.47194 2.587517e-09 8.311623e-06
## ENSG00000117152  6.628645  0.996321024 32.89300 9.737339e-09 2.606523e-05
## ENSG00000184564  6.275212  0.786215495 31.91089 1.614099e-08 3.703434e-05
## ENSG00000167759 -6.954394  1.078941938 29.29779 6.206642e-08 1.246061e-04
## ENSG00000165115 -8.928370  2.273562041 28.70860 8.412926e-08 1.360266e-04
## ENSG00000165046 -5.099561  0.288130401 28.41678 9.781367e-08 1.360266e-04</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw">decideTestsDGE</span>(res4))</code></pre></div>
<pre><code>##    [,1] 
## -1   292
## 0  15428
## 1    341</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSmear</span>(
    res4, <span class="dt">lowess =</span> <span class="ot">TRUE</span>,
    <span class="dt">de.tags =</span> <span class="kw">rownames</span>(<span class="kw">topTags</span>(res4, <span class="dt">n =</span> <span class="kw">sum</span>(<span class="kw">abs</span>(<span class="kw">decideTestsDGE</span>(res4))))$table)
)</code></pre></div>
<p><img src="15-remove-conf-reads_files/figure-html/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dealing-with-confounders.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="clustering-introduction.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/hemberg-lab/scRNA.seq.course/edit/master/15-remove-conf-reads.Rmd",
"text": "Edit"
},
"download": ["scRNA-seq-course.pdf"],
"toc": {
"collapse": "section"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
